GoogleForCreators/web-stories-wp
# Animation: Background Pan

## Summary
- Adds Initial background pan effect
- Sets up infra for dynamic disablable options in anim type inputs.

## Relevant Technical Choices
Mostly followed existing patterns. Lemme know if you see a better way to handle disabling options for anim inputs

## To-do


## User-facing changes
none, behind animation flag

## Testing Instructions
Add a pan animation effect to your background. see that it goes from the selected edge to where the background image was initially positioned.

Move background cropping around so that it touches an edge or is too small to pan in a direction. see that the pan options for those directions get disabled in both the direction input and the effect chooser.

---



Fixes #3529
